src/main/java/core/basesyntax/service/impl/TransactionProcessorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.TransactionProcessor;
+import core.basesyntax.strategy.OperationHandlerFactory;
+import java.util.List;
+
+public class TransactionProcessorImpl implements TransactionProcessor {
+    private final OperationHandlerFactory handlerFactory;
+
+    public TransactionProcessorImpl(OperationHandlerFactory handlerFactory) {
+        this.handlerFactory = handlerFactory;
+    }
+
+    @Override
+    public void process(List<Transaction> transactions) {
+        for (Transaction transaction : transactions) {
+            handlerFactory.getHandler(transaction.getOperation())
+                    .handle(transaction.getFruit().getName(), transaction.getQuantity());
+        }
+    }
+}
```

Comment:
The class `TransactionProcessorImpl` seems to be tightly coupled with the `OperationHandlerFactory`. According to the SOLID principles, and specifically the Dependency Inversion Principle, it would be better to depend on abstractions rather than concrete implementations. Consider introducing an interface for the factory or using a functional interface if applicable.

Reply:
The logic was changed